src/AndersonAcceleration.f90

```diff
@@ -6,55 +6,50 @@
 ! of Michigan and Oak Ridge National Laboratory.  The copyright and license    !
 ! can be found in LICENSE.txt in the head directory of this repository.        !
 !++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++!
-!> @brief Module provides an interface to Anderson Acceleration
-!>
-!> Currently supported TPLs include:
-!>  - Trilinos
-!>
-!> @par EXAMPLES
-!> @code
-!>
-!> @endcode
-!++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++!
+!> @brief A Fortran 2003 module that encompasses the native implementation of
+!>        Anderson Acceleration in Futility.
 MODULE AndersonAccelerationTypes
+#include "Futility_DBC.h"
+USE Futility_DBC
 USE IntrType
-USE BLAS
-USE trilinos_interfaces
-USE Times
-USE ExceptionHandler
+USE FutilityComputingEnvironmentModule
 USE ParameterLists
 USE ParallelEnv
 USE VectorTypes
+USE MatrixTypes
+USE LinearSolverTypes
+
 IMPLICIT NONE
 
 PRIVATE
-!
-! List of public members
-PUBLIC :: eAndersonAccelerationType
+
+!List of public members
 PUBLIC :: AndersonAccelerationType
 
-!> @brief the base eigenvalue solver type
+!> @brief the native Anderson Accelerator Type
 TYPE :: AndersonAccelerationType
   !> Initialization status
   LOGICAL(SBK) :: isInit=.FALSE.
-  !> NOX Solver type
-  INTEGER(SIK) :: id=-1
-  !> Pointer to the distributed memory parallel environment
-  TYPE(MPI_EnvType),POINTER :: MPIparallelEnv => NULL()
-  !> Pointer to the shared memory parallel environment TODO: eventually
-  TYPE(OMP_EnvType) :: OMPparallelEnv
-  !> iteration count
-  INTEGER(SIK) :: iter=0
-  !> size of nonlinear system
+  !> Size of solution vector/matrix
   INTEGER(SIK) :: n=-1
-  !> depth of anderson solver
+  !> Current iteration count
+  INTEGER(SIK) :: s=0
+  !> Iteration depth of Anderson solver
   INTEGER(SIK) :: depth=-1
-  !> value of mixing parameter
+  !> Starting iteration for Anderson
+  INTEGER(SIK) :: start=1
+  !> Value of mixing parameter
   REAL(SRK) :: beta=0.0_SRK
-  !> Pointer to solution vector, x
-  CLASS(VectorType),POINTER :: X=>NULL()
-  !> Timer to measure solution time
-  TYPE(TimerType) :: SolveTime
+  !> Initial iterates
+  REAL(SRK),ALLOCATABLE :: x(:,:)
```

Comment:
I thought we were changing the solver to use vector classes.

Reply:
We had talked about it, and I tried that. Switched to Vector classes and BLAS calls. Made the code slower (only 4% but still). Without an increase in performance I don't see much of a point in using vector classes here, so I switched back. Everything we want to accelerate in MPACT comes as some form/shape of REAL array (not vectors) which needs to be repacked into a 1D array. Any packing/repacking and the Anderson code itself is much cleaner when not using vector classes. Also, if anyone outside of the MPACT team wanted to use the Anderson ability not having to go through vector types/classes may help a beginner use this more easily.

Reply:
are any of the arrays more than 2 dimensional?

Reply:
Everything is changed over now.